### PR TITLE
deploy: fix initial image change deployments

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/9530

### CODE REVIEW

**Comment by kargakis** [comment]
[testextended][extended:core(Docker image source)]

**Comment by kargakis** [comment]
@mfojtik ptal

**Comment by kargakis** [comment]
green extended test [merge] @mfojtik review at your pleasure

File: pkg/deploy/controller/generictrigger/controller.go [hunk 1]
```diff
@@ -129,9 +129,12 @@ func canTrigger(config, decoded *deployapi.DeploymentConfig) (bool, []deployapi.
 
 		// We need stronger checks in order to validate that this template
 		// change is an image change. Look at the deserialized config's
-		// triggers and compare with the present trigger.
-		if !triggeredByDifferentImage(*t.ImageChangeParams, *decoded) {
-			continue
+		// triggers and compare with the present trigger. Initial deployments
+		// should always trigger since there is no previous config to compare to.
+		if config.Status.LatestVersion > 0 {
+			if !triggeredByDifferentImage(*t.ImageChangeParams, *decoded) {
```
Comment (mfojtik) [inline pkg/deploy/controller/generictrigger/controller.go]:
```go
if config.Status.LatestVersion > 0 && !triggeredByDifferentImage(*t.ImageChangeParams, *decoded) {
}
```

**Comment by mfojtik** [comment]
@kargakis one nit, LGTM

File: pkg/deploy/controller/generictrigger/controller.go (hunk 1 above)
Comment (kargakis) [inline pkg/deploy/controller/generictrigger/controller.go]: Will fix in https://github.com/openshift/origin/pull/9349 where I am moving all this code in rest.

Comment (mfojtik) [inline pkg/deploy/controller/generictrigger/controller.go]: ok.

**Comment by mfojtik** [comment]
@kargakis merge at green.

**Comment by kargakis** [comment]
Flaked on https://github.com/openshift/origin/issues/9444. [merge]

**Comment by kargakis** [comment]
[merge]

**Pull request merged**